applications/nrf_desktop/src/modules/bas.c

```diff
@@ -43,7 +43,8 @@ BT_GATT_SERVICE_DEFINE(bas_svc,
 	BT_GATT_CHARACTERISTIC(BT_UUID_BAS_BATTERY_LEVEL,
 			       BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
 			       BT_GATT_PERM_READ, read_blvl, NULL, &battery),
-	BT_GATT_CCC(blvl_ccc_cfg_changed),
+	BT_GATT_CCC(blvl_ccc_cfg_changed,
+		    BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
```

Comment:
I think this should be `BT_GATT_PERM_WRITE_ENCRYPT`, `BT_GATT_PERM_READ_ENCRYPT`.
I know that the original code has non-secured versions but the whole point of the upstream change was because we have reported these reports are read without protection. It would be good to set it right straight away :)

Reply:
As for the `BT_GATT_PERM_READ/WRITE_ENCRYPT` - should I update `bas` only, or `lbs` and `nus` as well?

Reply:
`bas.c` is part of `nrf_desktop`. I would prefer if it is updated as we need to do it anyway.
As for other services, as I commented, we don't use it so I leave it up to whoever owns them. Personally I would suggest making a kconfig option similar to HID so user could chose the level.

Reply:
Updated `bas.c` as requested.

Reply:
In fact Battery Service alone doesn't need these permissions to require link encryption. And in most cases it doesn't make sense to protect the access to battery level. It is only HOG spec that mandates this. So we can leave it for now, and we will add configurability later on.